### PR TITLE
[Feature] Modal: set width full on mobile and add hide header prop

### PR DESCRIPTION
**Pull request by Kelhel**
### ❓ Types of changes

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality)
- [ ] 📦 New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes
- [ ] The component exists on old Prestashop UIKit and my pull request on [migrating documentation](https://github.com/PrestaShopCorp/devdocs.uikit.prestashop.com) is accepted.

### CODE REVIEW

File: packages/components/modal/src/modal.ts [hunk 1]
```diff
@@ -69,6 +69,11 @@ export const modalProps = buildProps({
     required: false,
     default: undefined,
   },
+  hideHeader: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
```
Comment (kktos) [inline packages/components/modal/src/modal.ts:76]: I'd rather go for "positive" options like:
```typescript
  withHeader: {
    type: Boolean,
    required: false,
    default: true,
  },
```

**Review by kktos — CHANGES_REQUESTED** [review]

File: packages/components/modal/src/modal.ts (hunk 1 above)
Comment (MikeLebeau) [inline packages/components/modal/src/modal.ts:76]: Hm... ce serait pas mieux de juste rendre le prop `title` optional ? Parce que la ca veut dire que je suis forcé de donner un title et apres de mettre `hideHeader` a true c'est pas folichon

**Review by MikeLebeau — CHANGES_REQUESTED** [review]

File: packages/components/modal/src/modal.ts (hunk 1 above)
Comment (kktos) [reply packages/components/modal/src/modal.ts:76]: oh, nice catch indeed. the prop title is required.....
and that's KISS too.

Comment (Kelhel) [reply packages/components/modal/src/modal.ts]: Indeed, seems better as you suggest

**Review by kktos — APPROVED** [review]

**Review by MikeLebeau — APPROVED** [review]

**Pull request merged**